### PR TITLE
THREESCALE-10613 add userkey and applicationKey via cr

### PR DESCRIPTION
**Pull request by austincunningham**
# What
Issue [THREESCALE-10613](https://issues.redhat.com//browse/THREESCALE-10613)
add the ability to change UserKey and ApplicationKey via CR

# Verification
- install the operator following scripts will do this
[setup_db.sh](https://github.com/3scale-labs/3scale-perf-setup/blob/main/setup_db.sh)
[setup_apimanger.sh](https://github.com/3scale-labs/3scale-perf-setup/blob/main/setup_apimanager.sh)
[setup_product.sh](https://github.com/3scale-labs/3scale-perf-setup/blob/main/setup_product.sh)
- check the application UI to confirm that UserKey and ApplicationKey have been setup (will need to set the product to ApplcitionID & ApplictionKey in the product configuration page)
- scale down the operator deployment
- from this branch do a `make install` and `make run`
### Test generate secret
- Create an secret called auth-secret 
```bash
oc create secret generic auth-secret \
    --from-literal=UserKey=""\
    --from-literal=ApplicationKey=""\
    --namespace=3scale-test
```
- create a applicationAuth CR
```bash
oc apply -f - <<EOF
---
apiVersion: capabilities.3scale.net/v1beta1
kind: ApplicationAuth
metadata:
  name: applicationAuth-cr1
  namespace: 3scale-test
spec:
  applicationCRName: application-cr 
  generateSecret: true
  authSecretRef: 
     name: auth-secret
EOF
```
- confirm in the application ui that the UserKey and ApplicationKey(should see 2 keys) have updated 
- confirm that the auth-secret values have updated to match the latest keys UI and that the ApplicationID is present
### Test manual secret 
- Create an secret called auth-secret2 with the following values
```bash
oc create secret generic auth-secret2 \
    --from-literal=UserKey="testUserKey"\
    --from-literal=ApplicationKey="testApplicationKey"\
    --namespace=3scale-test
```
- create a applicationAuth CR with out genetate secret
```bash
oc apply -f - <<EOF
---
apiVersion: capabilities.3scale.net/v1beta1
kind: ApplicationAuth
metadata:
  name: applicationAuth-cr2
  namespace: 3scale-test
spec:
  applicationCRName: application-cr 
  authSecretRef: 
     name: auth-secret2
EOF
```
- Outcome in the application UI is that the UserKey and ApplictionKey will be populated with the values from the secret 
- The ApplicationID will be added to the auth-secret2

### CODE REVIEW

File: doc/operator-application-capabilities.md [hunk 1]
```diff
@@ -1643,6 +1715,132 @@ status:
...
+metadata:
+  name: auth-secret
+stringData:
+  UserKey: testUserKey
+  ApplicationKey: testApplicationKey
+  ApplicationID: <ApplicationID value>
+```
+[ApplicationAuth CRD reference](applicationauth-reference.md) for more info about fields.
+
+### Application Custom Resource Status Fields
```
Comment (carlkyrillos) [inline doc/operator-application-capabilities.md:1810]:
```suggestion
### ApplicationAuth Custom Resource Status Fields
```
Typo

File: doc/operator-application-capabilities.md [hunk 2]
```diff
@@ -66,16 +68,20 @@ The following diagram shows available custom resource definitions and their rela
       * [Create developer user with admin role](#create-developer-user-with-admin-role)
       * [DeveloperUser custom resource status field](#developeruser-custom-resource-status-field)
       * [Link your DeveloperUser to your 3scale tenant or provider account](#link-your-developeruser-to-your-3scale-tenant-or-provider-account)
-   * [Application Custom Resource](#application-custom-resource)
-      * [Application Custom Resource Status Fields](#application-custom-resource-status-fields)
+   * [Application custom resource](#application-custom-resource)
+      * [Application custom resource status fields](#application-custom-resource-status-fields)
       * [Application Misconfiguration Errors](#application-misconfiguration-errors)
+   * [ApplicationAuth custom resource](#applicationauth-custom-resource)
+      * [Application custom resource status fields](#applicationauth-custom-resource-status-fields)
```
Comment (carlkyrillos) [inline doc/operator-application-capabilities.md:75]:
```suggestion
      * [ApplicationAuth custom resource status fields](#applicationauth-custom-resource-status-fields)
```
Typo

File: go.mod [hunk 3]
```diff
@@ -3,7 +3,7 @@ module github.com/3scale/3scale-operator
 go 1.19
 
 require (
-	github.com/3scale/3scale-porta-go-client v0.10.0
+	github.com/3scale/3scale-porta-go-client v0.9.1-0.20240208161959-a6c3b3f36473
```
Comment (carlkyrillos) [inline go.mod:6]: @austincunningham Do we need to downgrade here?

**Review by carlkyrillos — COMMENTED** [review]
Completed the verification steps and left a few docs comments but the actual code changes look good to me.

@austincunningham I think you need to rebase off of master because there have been a few changes to main.go (one [example](https://github.com/carlkyrillos/3scale-operator/blob/master/main.go#L84-L90)) that this branch is missing. I'm not sure why they aren't being highlighted/conflicted

File: go.mod (hunk 3 above)
Comment (austincunningham) [reply go.mod:6]: hmm yea I was working of a commit sha for porta so if v0.10.0 has what I need then no not needed.  Suspect I need a release of 3scale-porta-client.

Comment (austincunningham) [reply go.mod:6]: yep need a release 
image

**Review by carlkyrillos — APPROVED** [review]
Reviewed latest changes
/lgtm

**Pull request merged**